#!/usr/bin/env python
PACKAGE = "hanse_pipefollowing"
import roslib;roslib.load_manifest(PACKAGE)

from math import pi

from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()

gen.add("minSize", double_t, 0, "", 0.05, 0.0, 1.0)
gen.add("maxSize", double_t, 0, "", 0.4, 0.0, 1.0)
gen.add("fwSpeed", double_t, 0, "", 0.8, 0.1, 1.0)
gen.add("deltaAngle", double_t, 0, "in radians", 0.192, 0.01, pi/4)
gen.add("deltaDist", double_t, 0, "", 100, 0, 600)
gen.add("kpAngle", double_t, 0, "", 1.0, 0.0, 5.0)
gen.add("kpDist", double_t, 0, "", 1.0, 0.0, 5.0)
gen.add("robCenterX", double_t, 0, "", 320, 0.0, 640)
gen.add("robCenterY", double_t, 0, "", 248, 0.0, 480)
gen.add("maxDistance", double_t, 0, "", 320, 0.0, 600)


exit(gen.generate(PACKAGE, "pipefollowing", "PipeFollowing"))
